lib/cli-engine.js

```diff
@@ -575,11 +585,27 @@ CLIEngine.prototype = {
          * Executes the linter on a file defined by the `filename`. Skips
          * unsupported file extensions and any files that are already linted.
          * @param {string} filename The resolved filename of the file to be linted
+         * @param {boolean} warnIgnored always warn when a file is ignored
          * @returns {void}
          */
-        function executeOnFile(filename) {
+        function executeOnFile(filename, warnIgnored) {
             var hashOfConfig;
 
+            if ((options.ignore !== false) && ignoredPaths.contains(filename, "custom")) {
```

Comment:
Can you combine this and the `if` on 601 into a single block that checks for `options.ignore !== false`.